--- Pet.cpp	2018-09-22 22:07:12.923347056 +0000
+++ Pet.cpp.beastmaster	2018-10-13 22:33:16.325393243 +0000
@@ -104,8 +104,10 @@
         return false;
 
     // DK Pet exception
+    /* no exception due to mod-npcbeastmaster
     if (owner->getClass() == CLASS_DEATH_KNIGHT && !owner->CanSeeDKPet())
         return false;
+    */
 
     uint32 ownerid = owner->GetGUIDLow();
     PreparedStatement* stmt;
@@ -673,6 +675,7 @@
     PetType petType = MAX_PET_TYPE;
     if (IsPet() && m_owner->GetTypeId() == TYPEID_PLAYER)
     {
+        /* use another check due to mod-npcbeastmaster
         if (m_owner->getClass() == CLASS_WARLOCK ||
             m_owner->getClass() == CLASS_SHAMAN ||          // Fire Elemental
             m_owner->getClass() == CLASS_DEATH_KNIGHT ||    // Risen Ghoul
@@ -685,6 +688,17 @@
         }
         else
             sLog->outError("Unknown type pet %u is summoned by player class %u", GetEntry(), m_owner->getClass());
+        */
+        // better pet handling for mod-npcbeastmaster
+        if (cinfo->IsTameable(true))
+        {
+            petType = HUNTER_PET;
+            m_unitTypeMask |= UNIT_MASK_HUNTER_PET;
+        }
+        else
+        {
+            petType = SUMMON_PET;
+        }
     }
 
     uint32 creature_ID = (petType == HUNTER_PET) ? 1 : cinfo->Entry;
